state/execution.go

```diff
@@ -924,12 +947,18 @@ func (blockExec *BlockExecutor) saveFailedProposalBlock(state State, block *type
 	}
 
 	debugDir := filepath.Join(blockExec.rootDir, "data", "debug")
-	filename := fmt.Sprintf("%s-%d-%s_failed_proposal.pb",
+	baseFilename := fmt.Sprintf("%s-%d-%s_failed_proposal.pb",
 		state.ChainID,
 		block.Height,
 		reason,
 	)
 
+	filename, err := FindNextAvailableFilename(debugDir, baseFilename)
+	if err != nil {
+		blockExec.logger.Error("failed to find available filename for failed proposal block", "err", err.Error(), "reason", reason)
+		return
+	}
```

Comment:
we at least need to change the title of this PR since we're not adding round

if we just need an incremental value then we just add the millisecond timestamp and call it good. In process proposal I think we're adding a round, but ultimately this doesn't really add that much value since a proposer would need to propose multiple invalid blocks to get hit. using a timestamp would provide an equal amount of value since we'd know which invalid proposals came later even if it would get hit.